tests/boot/with_mcumgr/pytest/mcumgr.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import logging
+import re
+
+from subprocess import check_output
+from pathlib import Path
+
+
+logger = logging.getLogger(__name__)
+
+
+class McuMgrException(Exception):
+    """General mcumgr exception."""
+
+
+class McuMgr:
+    """Sample wrapper for mcumgr command-line tool"""
+    mcumgr_exec = 'mcumgr'
+
+    def __init__(self, connection_options: str):
+        self.conn_opts = connection_options
+
+    @classmethod
+    def create_for_serial(cls, serial_port: str) -> McuMgr:
+        return cls(connection_options=f'--conntype serial --connstring={serial_port}')
+
+    def run_command(self, cmd: str, log_output=False) -> str:
+        command = f'{self.mcumgr_exec} {self.conn_opts} {cmd}'
+        logger.info(f"CMD: {command}")
+        return check_output(command, shell=True, text=True)
+
+    def reset_device(self):
+        self.run_command('reset')
+
+    def image_upload(self, image: Path | str):
+        self.run_command(f'-t 60 image upload {image}')
```

Comment:
60 seconds is a long time for a packet timeout, 25 would be more than sufficient